modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/set_permissions_command.rb

```diff
@@ -45,25 +45,33 @@ def delete? = permission_ids.any? && user_ids.empty?
             def update? = permission_ids.any? && user_ids.any?
           end
 
-          def self.call(storage:, auth_strategy:, path:, permissions:)
-            new(storage).call(auth_strategy:, path:, permissions:)
+          def self.call(storage:, auth_strategy:, input_data:)
+            new(storage).call(auth_strategy:, input_data:)
           end
 
           def initialize(storage)
             @storage = storage
           end
 
-          def call(auth_strategy:, path:, permissions:)
+          # rubocop:disable Metrics/AbcSize
+          def call(auth_strategy:, input_data:)
             with_tagged_logger do
+              info "Validating input data"
+              validate_input_data(input_data).on_failure { return _1 }
```

Comment:
🔴 We probably should move this to the `.call` method, so we don't even instantiate the service if `InputData` is invalid.

Reply:
Well, this would work, if we make `#call` private. And only execute the command with `.call`. Should we change that behaviour?

Reply:
validation is now handled outside